### PR TITLE
{Compute} `az vm host group` and `az vmss nic` and `az sig gallery-application`: Migrate to atomic commands

### PR DESCRIPTION
**Pull request by yanzhudd**
**Related command**

`az vm host group list/show/delete`
`az vmss nic list/show/list-vm-nics`
`az sig gallery-application list/show/delete`

**Description**

Migrate part of `az vm host group` and `az sig gallery-application` commands. Migrate `az vmss nic` all commands.

Some commands in `az vm host group` are not migrated yet:

`az vm host group create`
`az vm host group get-instance-view`
`az vm host group update`

Some commands in `az sig gallery-application` are not migrated yet:

`az sig gallery-application create`
`az sig gallery-application update`

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/aaz/2018_03_01_hybrid/vmss/nic/_list.py [hunk 1]
```diff
@@ -0,0 +1,800 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "vmss nic list",
+)
+class List(AAZCommand):
+    """Gets all network interfaces in a virtual machine scale set.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/2018_03_01_hybrid/vmss/nic/_list.py:18]:
```suggestion
    """Get all network interfaces in a virtual machine scale set.
```

File: src/azure-cli/azure/cli/command_modules/vm/aaz/2018_03_01_hybrid/vmss/nic/_list_vm_nics.py [hunk 2]
```diff
@@ -0,0 +1,809 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "vmss nic list-vm-nics",
+)
+class ListVmNics(AAZCommand):
+    """Gets information about all network interfaces in a virtual machine in a virtual machine scale set.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/2018_03_01_hybrid/vmss/nic/_list_vm_nics.py:18]:
```suggestion
    """Get information about all network interfaces in a virtual machine in a virtual machine scale set.
```

File: src/azure-cli/azure/cli/command_modules/vm/aaz/2019_03_01_hybrid/vm/host/group/_list.py [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "vm host group list",
+)
+class List(AAZCommand):
+    """List dedicated host groups.
+
+    Lists dedicated host groups by subscription. If resource group is specified, lists dedicated host groups by resource group.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/2019_03_01_hybrid/vm/host/group/_list.py:20]:
```suggestion
    List dedicated host groups by subscription. If resource group is specified, list dedicated host groups by resource group.
```

File: src/azure-cli/azure/cli/command_modules/vm/aaz/2019_03_01_hybrid/vmss/nic/_list.py [hunk 4]
```diff
@@ -0,0 +1,800 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "vmss nic list",
+)
+class List(AAZCommand):
+    """Gets all network interfaces in a virtual machine scale set.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/2019_03_01_hybrid/vmss/nic/_list.py:18]:
```suggestion
    """Get all network interfaces in a virtual machine scale set.
```

File: src/azure-cli/azure/cli/command_modules/vm/aaz/latest/sig/gallery_application/version/_delete.py [hunk 5]
```diff
@@ -16,7 +16,7 @@
     confirmation="Are you sure you want to perform this operation?",
 )
 class Delete(AAZCommand):
-    """Delete a gallery application version.
+    """Delete a gallery Application Version.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/latest/sig/gallery_application/version/_delete.py:19]: Why `Application Version` starts with capital letters?

File: src/azure-cli/azure/cli/command_modules/vm/aaz/latest/sig/gallery_application/version/_show.py [hunk 6]
```diff
@@ -15,7 +15,7 @@
     "sig gallery-application version show",
 )
 class Show(AAZCommand):
-    """Get information about a gallery application version.
+    """Get information about a gallery Application Version.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/latest/sig/gallery_application/version/_show.py:18]: Same as above

**Comment by zhoxing-ms** [comment]
The `vm host` related commands don't seem to have tests in the old profiles. Could we also add the `test_vm_host_management` to the old profiles?

**Comment by yanzhudd** [comment]
> The `vm host` related commands don't seem to have tests in the old profiles. Could we also add the `test_vm_host_management` to the old profiles?

Okay. I have added three tests related to `vm host group` in 2020-09-01-hybrid profile.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**